src/react/actions/account.js

```diff
@@ -103,3 +107,61 @@ export function deleteAccount(accountName: string): ThunkStatePromisedAction {
             });
     };
 }
+
+export function listAccountAccessKeys(accountID: string): ThunkStatePromisedAction {
+    return (dispatch: DispatchFunction, getState: GetStateFunction) => {
+        const state = getState();
+        const { oidc } = state;
+        const { stsClient, iamClient } = getClients(state);
+
+        const assumeRoleParams = {
+            idToken: oidc.user.id_token,
+            RoleSessionName:'app1',
+            roleArn: `arn:aws:iam::${accountID}:role/roleForB`,
+        };
+        dispatch(networkStart('Listing account access keys'));
+        return stsClient.assumeRoleWithWebIdentity(assumeRoleParams)
+            .then(creds => {
+                const params = {
+                    accessKey: creds.Credentials.AccessKeyId,
+                    secretKey: creds.Credentials.SecretAccessKey,
+                    sessionToken: creds.Credentials.SessionToken,
+                };
+                iamClient.login(params);
+                return iamClient.listOwnAccessKeys();
+            })
+            .then(resp => dispatch(updateAccessKeysList(resp.AccessKeyMetadata)))
+            .catch(error => dispatch(handleClientError(error)))
+            .catch(error => dispatch(handleApiError(error, 'byModal')))
+            .finally(() => dispatch(networkEnd()));
+    };
+}
+
+export function deleteAccountAccessKey(accessKey: string, accountID: string): ThunkStatePromisedAction {
+    return (dispatch: DispatchFunction, getState: GetStateFunction) => {
+        const state = getState();
+        const { oidc } = state;
+        const { stsClient, iamClient } = getClients(state);
+
+        const assumeRoleParams = {
+            idToken: oidc.user.id_token,
+            RoleSessionName:'app1',
+            roleArn: `arn:aws:iam::${accountID}:role/roleForB`,
+        };
+        dispatch(networkStart('Deleting account access key'));
+        return stsClient.assumeRoleWithWebIdentity(assumeRoleParams)
+            .then(creds => {
+                const params = {
+                    accessKey: creds.Credentials.AccessKeyId,
+                    secretKey: creds.Credentials.SecretAccessKey,
+                    sessionToken: creds.Credentials.SessionToken,
+                };
+                iamClient.login(params);
```

Comment:
Let's keep the stored clients in sync with the account context. Meaning, here let's instantiate a new iamClient and not retrieve/modify the stored one.